test/unit_test/deepdoc/parser/test_get_text_encoding.py

```
@@ -0,0 +1,122 @@
+from unittest import mock
+
+import pytest
+
+# Import get_text directly by file path so the test does not depend on the heavy
+# transitive imports pulled in by deepdoc/parser/__init__.py (pypdf, etc.).
+# find_codec is the one real dependency the function exercises: use the genuine
+# implementation when the full environment is available, otherwise fall back to
+# an equivalent chardet-based detector so the test still runs in a minimal lane.
+try:
+    from rag.nlp import find_codec as _real_find_codec
+except Exception:  # pragma: no cover - only taken in a stripped-down env
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)